torch/distributed/checkpoint/default_planner.py

```diff
@@ -250,9 +250,16 @@ def create_default_local_load_plan(
     It handles resharding by issuing multiple read requests against storage in order to match
     load requirements.
     """
+
     for fqn, obj in state_dict.items():
         md = metadata.state_dict_metadata[fqn]
-        requests += _create_read_items(fqn, md, obj)
+        # Since DTensor supports submesh, adding extra check to ensure _create_read_items()
+        # gets called only when the current rank is part of the mesh for the corresponding DTensor.
+        if isinstance(obj, DTensor):
+            if obj.device_mesh.get_coordinate() is not None:
```

Comment:
curious if dcp actually writes dtensor's metadata to disk or no? i.e. device_mesh, placements, etc.

Reply:
Right now, the metadata looks like this. https://github.com/pytorch/pytorch/blob/master/torch/distributed/checkpoint/planner_helpers.py#L68 

The metadata was originally built around ShardedTensor, which we will probably have to revisit later on.